File: q1200/q1200.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int min_diff = INT_MAX;
        for (int i = 0; i < arr.size()-1; i++){
            min_diff = min(min_diff, arr[i+1]-arr[i]);
        }

        vector<vector<int>> res;
        for (int i = 0; i < arr.size()-1; i++){
            if (arr[i+1]-arr[i] == min_diff){
                res.push_back({arr[i], arr[i+1]});
            }
        }
        
        return res;
    }
};

File: q1200/q1200.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        min_diff = float("inf")
        for i in range(len(arr) - 1):
            min_diff = min(min_diff, arr[i + 1] - arr[i])

        res = []
        for i in range(len(arr) - 1):
            if arr[i + 1] - arr[i] == min_diff:
                res.append([arr[i], arr[i + 1]])

        return res
